src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -112,24 +112,28 @@ object Template {
   * Utility methods for working with BAMs.
   */
 object Bams extends LazyLogging {
+
+  /** The default maximum reads in memory for all sorting collections when no `maxInMemory` option is given. */
+  val MaxReadsInMemory: Int = 1e6.toInt
+
   /**
     * Builds a sorting collection for sorting SAMRecords.
     *
     * @param order the desired output sort order
     * @param header the header to use to encode records during sorting
-    * @param maxInMemory the maximum number of records to keep and sort in memory
+    * @param maxInMemory optionally the maximum number of records to keep and sort in memory
     * @param tmpDir an optional temp directory to use for temporary sorting files
     * @return a SortingCollection
     */
   def sortingCollection(order: SortOrder,
                         header: SAMFileHeader,
-                        maxInMemory: Int = 1e6.toInt,
+                        maxInMemory: Option[Int] = None,
```

Comment:
I don't like this change.  What does it mean to set it to `None` exactly?  That there will be none kept in memory?  Since you immediately substitute in the default value, what is the point of this change other than obfuscating that there is a default?